api/controller/pokemon_controller.go

```diff
@@ -0,0 +1,36 @@
+package controller
+
+import (
+	"go-bootcamp/service"
+	"strconv"
+
+	"github.com/gin-gonic/gin"
+	"github.com/pkg/errors"
+)
+
+type PokemonController interface {
+	GetPokemonByID(c *gin.Context)
+}
+
+type pokemonController struct {
+	pokemonService service.PokemonService
+}
+
+func NewPokemonController(pokemonService service.PokemonService) PokemonController {
+	return &pokemonController{pokemonService: pokemonService}
+}
+
+func (ctrl *pokemonController) GetPokemonByID(c *gin.Context) {
+	idParam := c.Param("id")
+	id, err := strconv.Atoi(idParam)
+	if err != nil {
+		c.JSON(400, gin.H{"error": errors.Wrap(err, "invalid ID").Error()})
```

Comment:
You can use the `net/http` package, it contains constants for HTTP stuff.
 Instead of using magic numbers you can use the constants
 `400` -> `http.StatusNotFound`

Reply:
changed the numbers for the http status